File: Codechef/Chef_Apps/Chef_Apps.cpp
#include <iostream>
using namespace std;


int main()
{
    int T;
    cin >> T;
    
    while (T--)
    {
        int S, X, Y, Z, remain;
        cin >> S >> X >> Y >> Z;
        if (Z <= remain)
        {
            cout << "0" << endl;
        }
        else if (S >= (X + Z) || Z >= (Y + Z))
        {
            cout << "1" << endl;
        }
        else
        {
            cout << "2" << endl;
        }

    }
    return 0;
}

File: Codechef/Chef_Apps/Chef_Apps.py
T = int(input())
while T:
    T -= 1
    S, X, Y, Z = map(int, input().split())
    remain = S - (X + Y)
    if Z <= remain:
        print(0)
    elif (S >= X + Z or S >= Y + Z):
        print(1)
    else:
        print(2)